app/src/main/java/com/keylesspalace/tusky/BaseActivity.java

```diff
@@ -93,6 +97,19 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         requesters = new HashMap<>();
     }
 
+    @Override
+    protected void attachBaseContext(Context newBase) {
+        Log.d(TAG, "attachBaseContext()");
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(newBase);
+
+        float uiScaleRatio = preferences.getFloat(PrefKeys.UI_SCALE_RATIO, 100F);
+        Context context = FontScaleContextWrapper.Companion.wrap(
+                newBase,
+                uiScaleRatio / 100.f
```

Comment:
Curiosity: This takes care of "all" ui texts?

Reply:
I believe so. The Context carries the font scale ratio, and that's used to compute the text size in pixels from dp, sp, etc.

This is the same mechanism Android uses if you use the system settings to adjust the font scale device wide.